### PR TITLE
feat: add param for persistent db

### PR DESCRIPTION
**Pull request by imsitnikov**

### CODE REVIEW

File: docker-compose.yml [hunk 1]
```diff
@@ -68,13 +67,15 @@ services:
...
-      POSTGRES_DSN_LIST: "postgres://us:us@pg-us:5432/us-db-ci_purgeable"
+      POSTGRES_DSN_LIST: ${US_POSTGRES_DSN_LIST:-postgres://us:us@pg-us:5432/us-db-ci_purgeable}
       MASTER_TOKEN: "fake-us-master-token"
       PG_TEST_DATA: 1
       CH_DEMO_DATA: 1
       HC: ${HC}
     ports:
       - "50500:80"
+    extra_hosts:
+      - "host.docker.internal:host-gateway"
```
Comment (imsitnikov) [inline docker-compose.yml:78]: `host.docker.internal` is an alias for `localhost` on docker host machine. For cases when db is located there.

**Review by Konstantin-Roumas — APPROVED** [review]

File: README.md [hunk 2]
```diff
@@ -20,7 +20,12 @@ Use the following command to start DataLens containers:
 
 ```bash
 git clone https://github.com/datalens-tech/datalens && cd datalens
+
+# with demo database
 HC=1 docker compose up
+
+# with persistent database
```
Comment (resure) [inline README.md:27]: we will use persistent database by default (#22), so we should choose a different word here. Maybe "external" database?

Comment (imsitnikov) [reply README.md]: done

File: README.md [hunk 3]
```diff
@@ -25,15 +25,7 @@ git clone https://github.com/datalens-tech/datalens && cd datalens
...
-US_POSTGRES_DSN_LIST="postgres://..." HC=1 docker compose up
-```
-
-Use the additional parameter `US_SKIP_INSTALL_DB_EXTENSIONS=1` if the external database user doesn't have access to install extensions. In this case, you should manually install extensions before launching the app:
-```sql
-CREATE EXTENSION IF NOT EXISTS pg_trgm;
-CREATE EXTENSION IF NOT EXISTS btree_gin;
-CREATE EXTENSION IF NOT EXISTS btree_gist;
-CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+US_POSTGRES_DSN_LIST="postgres://{user}:{password}@{host}:{port}/{database}?ssl=true" HC=1 docker compose up
```
[the remark this answers is not in the document]
Comment (imsitnikov) [reply README.md]: removed `ssl=true`

File: README.md [hunk 4]
```diff
@@ -63,4 +55,25 @@ We are releasing DataLens with first minimal set of available connectors (clickh
...
+
+This may be caused because of external database user doesn't have permission to install extensions. Use the additional parameter `US_SKIP_INSTALL_DB_EXTENSIONS=1` to skip this step, but in this case you should manually install extensions before launching the app:
+
+```sql
+CREATE EXTENSION IF NOT EXISTS pg_trgm;
+CREATE EXTENSION IF NOT EXISTS btree_gin;
+CREATE EXTENSION IF NOT EXISTS btree_gist;
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+```
+
```
[the remark this answers is not in the document]
Comment (imsitnikov) [reply README.md:70]: added

File: README.md [hunk 5]
```diff
@@ -50,4 +55,25 @@ We are releasing DataLens with first minimal set of available connectors (clickh
 
 **Where can I find persistent application data storage?**
 
-We use the `.us-data` folder to store PostgreSQL data permanently. You can delete this folder if you want, it will be recreated with the demo data after restarting the `datalens-us` container.
\ No newline at end of file
+We use the `.us-data` folder to store PostgreSQL data permanently. You can delete this folder if you want, it will be recreated with the demo data after restarting the `datalens-us` container.
+
+**I use external database and `datalens-us` container does not start.**
+
+This may be caused because of external database user doesn't have permission to install extensions. Use the additional parameter `US_SKIP_INSTALL_DB_EXTENSIONS=1` to skip this step, but in this case you should manually install extensions before launching the app:
```
[the remark this answers is not in the document]
Comment (imsitnikov) [reply README.md]: changed

File: README.md [hunk 6]
```diff
@@ -20,7 +20,12 @@ Use the following command to start DataLens containers:
...
 git clone https://github.com/datalens-tech/datalens && cd datalens
+
+# with demo database
 HC=1 docker compose up
+
+# with external database
+US_POSTGRES_DSN_LIST="postgres://{user}:{password}@{host}:{port}/{database}?ssl=true" HC=1 docker compose up
 ```
 
 This command will launch all containers required to run DataLens and UI will be available on http://localhost:8080
```
[the remark this answers is not in the document]
Comment (imsitnikov) [reply README.md:31]: fixed

**Comment by imsitnikov** [comment]
@resure I've changed description of the `METADATA_SKIP_INSTALL_DB_EXTENSIONS` param.

File: README.md [hunk 7]
```diff
@@ -44,10 +49,35 @@ DataLens consists of the three main parts:
...
+We use some PostgresSQL extensions for the metadata database and the application checks for them on startup and tries to install them if they are not already installed. Check your database user's rights for installing extensions by trying to install them manually:
+
+```sql
+CREATE EXTENSION IF NOT EXISTS pg_trgm;
+CREATE EXTENSION IF NOT EXISTS btree_gin;
+CREATE EXTENSION IF NOT EXISTS btree_gist;
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
+```
+
+If this attempt is unsuccessful, try to install dependencies by database admin and add param `METADATA_SKIP_INSTALL_DB_EXTENSIONS=1` on startup, this parameter allows app to skip installing extensions.
```
Comment (Marginy605) [inline README.md:71]: `allows app` looks like missed article

File: README.md [hunk 8]
```diff
@@ -44,10 +49,35 @@ DataLens consists of the three main parts:
...
+
+If you're using managed database it's also possible that extensions for your database cluster are controlled by external system and could be changed only using it's UI or API. In that case, consult with documentation for managed database service that you're using. Don't forget to add `METADATA_SKIP_INSTALL_DB_EXTENSIONS=1` after installing extensions this way.
```
Comment (Marginy605) [inline README.md:73]: comma after `database`
`In that case` ->  better `In this case` or `In such case`
`that you're using` -> `which you're using`

File: README.md [hunk 9]
```diff
@@ -44,10 +49,35 @@ DataLens consists of the three main parts:
...
+
+**My PostgresSQL cluster has multiple hosts, how can I specify them in `METADATA_POSTGRES_DSN_LIST` param?**
+
+Yes, you can write all cluster hosts separated by commas:
```
Comment (Marginy605) [inline README.md:77]: It looks inconsistent. The question is "how can I" and the answer is "Yes, you can". It's better to change form of the question or the answer.

File: README.md [hunk 10]
```diff
@@ -44,10 +49,35 @@ DataLens consists of the three main parts:
...
+
+`METADATA_POSTGRES_DSN_LIST="postgres://{user}:{password}@{host_1}:{port}/{database},postgres://{user}:{password}@{host_2}:{port}/{database},postgres://{user}:{password}@{host_3}:{port}/{database}" ...`
+
+**How can I specify custom certificate for connecting to metadata database?**
 
-We use the `.us-data` folder to store PostgreSQL data permanently. You can delete this folder if you want, it will be recreated with the demo data after restarting the `datalens-us` container.
\ No newline at end of file
+You can add additional certificates for database to `./certs/root.crt`, they will be used for connecting to database from the `datalens-us` container.
```
Comment (Marginy605) [inline README.md:83]: `for database to` -> it's better `to database in`
`they will be used for connecting to database` -> `they will be used to connect to the database` or `they will be used for connection[s] to the database`

File: README.md [hunk 11]
```diff
@@ -44,10 +49,35 @@ DataLens consists of the three main parts:
...
+We are releasing DataLens with first minimal set of available connectors (clickhouse, clickhouse over ytsaurus and postgresql) as well as other core functionality such as data processing engine and user interface. However, to kick off this project in a reasonable timeframe we have chosen to drop some of the features out of the first release: this version does not contain middleware and components for user sessions, object ACLs and multitenancy (although code contains entry-points for such extensions). We are planning to add missing features based on our understanding of community priorities and your feedback.
 
 ## FAQ
 
-**Where can I find persistent application data storage?**
+**Where does DataLens store it's metadata?**
+
+We use the `metadata` folder to store PostgreSQL data. If you want to start over, you can delete this folder: it will be recreated with demo objects on the next start of the `datalens-us` container.
+
+**I use the `METADATA_POSTGRES_DSN_LIST` param for external metadata database and app does not start. What could be the reason?**
```
Comment (Marginy605) [inline README.md:60]: `and app does not start` -> `and the app doesn't start`

File: README.md [hunk 12]
```diff
@@ -44,10 +49,35 @@ DataLens consists of the three main parts:
...
+
+We use some PostgresSQL extensions for the metadata database and the application checks for them on startup and tries to install them if they are not already installed. Check your database user's rights for installing extensions by trying to install them manually:
```
Comment (Marginy605) [inline README.md:62]: `checks for them on startup` -> `checks them at startup`
`if they are not already installed` -> `if they haven't been already installed`

**Review by Marginy605 — CHANGES_REQUESTED** [review]

File: README.md (hunk 8 above)
Comment (imsitnikov) [reply README.md]: done

File: README.md (hunk 9 above)
Comment (imsitnikov) [reply README.md]: changed:
`Yes, you can write` -> `You can write`

File: README.md (hunk 10 above)
Comment (imsitnikov) [reply README.md]: fixed

File: README.md (hunk 12 above)
Comment (imsitnikov) [reply README.md]: changed

**Review by Marginy605 — APPROVED** [review]

**Review by resure — APPROVED** [review]

**Pull request merged**